pymongo/asynchronous/mongo_client.py

```diff
@@ -750,6 +755,9 @@ def __init__(
             port = self.PORT
         if not isinstance(port, int):
             raise TypeError(f"port must be an instance of int, not {type(port)}")
+        self._host = host
+        self._port = port
+        self._topology: Optional[Topology] = None
```

Comment:
Adding `assert self._topology is not None` so many times feels a bit off since it adds runtime overhead for a static typing issue. What if we do this instead?:
```python
        self._topology: Topology = None  # type: ignore[assignment]
```